client/app/src/services/network.service.js

```diff
@@ -354,25 +354,83 @@
     // Updates peer with all currency values relative to the USD price.
     function updatePeerWithCurrencies (peer, res) {
       let deferred = $q.defer()
-      const currencies = ['AUD', 'BRL', 'CAD', 'CHF', 'CNY', 'EUR', 'GBP', 'HKD', 'IDR', 'INR', 'JPY', 'KRW', 'MXN', 'RUB']
-      var currencyConversionRequest = createCurrencyConversionApiCall(currencies)
-      $http.get(currencyConversionRequest, {timeout: 2000}).then(function (result) {
-        const USD_PRICE = Number(res.data[0].price_usd)
-        var prices = {}
-        currencies.forEach(function (currency) {
-          prices[currency.toLowerCase()] = result.data.rates[currency] * USD_PRICE
-        })
-        prices['btc'] = res.data[0].price_btc
-        prices['usd'] = res.data[0].price_usd
-        storageService.setGlobal('peerCurrencies', prices)
-        deferred.resolve(prices)
-      }, () => {
-        deferred.reject(false)
+      peer = updateCurrencyConversionRates(peer)
+      const USD_PRICE = Number(res.data[0].price_usd)
+      var currencies = ['AUD', 'BRL', 'CAD', 'CHF', 'CNY', 'EUR', 'GBP', 'HKD', 'IDR', 'INR', 'JPY', 'KRW', 'MXN', 'RUB']
+      var prices = {}
+      currencies.forEach(function (currency) {
+        prices[currency.toLowerCase()] = peer.market.conversionRates[currency] * USD_PRICE
       })
-
+      prices['btc'] = res.data[0].price_btc
+      prices['usd'] = res.data[0].price_usd
+      storageService.setGlobal('peerCurrencies', prices)
+      deferred.resolve(prices)
       return deferred.promise
     }
 
+    // Updates the currency conversion rates IF necessary
+    // Necessary if it isn't stored, or if the stored value is too old
+    function updateCurrencyConversionRates (peer) {
+      if (storageService.getGlobal('conversionRates') !== undefined) {
+        var priceObj = storageService.getGlobal('conversionRates')
```

Comment:
We should probably utilize 'let' instead of 'var' unless there is a specific use case that doesn't allow it